src/i_video.c

```diff
@@ -665,9 +665,27 @@ void I_StartFrame(void)
 
 static void UpdateRender(void)
 {
+    void *pixels;
+    int pitch;
+
+    if (SDL_LockTexture(texture, NULL, &pixels, &pitch) != 0)
```

Comment:
Why not `&blit_rect` here?

Reply:
I confirm small performance improvements. `SDL_UpdateTexture` doesn't work properly with direct3d9 renderer for some reason? Not sure.
Anyway, I suggest this changes:
```diff
diff --git a/src/i_video.c b/src/i_video.c
index ae7e26e8..ae672e86 100644
--- a/src/i_video.c
+++ b/src/i_video.c
@@ -668,19 +668,12 @@ static void UpdateRender(void)
     void *pixels;
     int pitch;
 
-    if (SDL_LockTexture(texture, NULL, &pixels, &pitch) != 0)
-    {
-        I_Error("Failed to lock texture: %s", SDL_GetError());
-    }
+    SDL_LockTexture(texture, &blit_rect, &pixels, &pitch);
 
-    if (!argbbuffer || argbbuffer->pixels != pixels)
+    if (!argbbuffer)
     {
-        I_Printf(VB_DEBUG, "Creating blit surface");
-
-        SDL_FreeSurface(argbbuffer);
-
         argbbuffer = SDL_CreateRGBSurfaceWithFormatFrom(
-            pixels, screenbuffer->w, screenbuffer->h, 8,
+            pixels, blit_rect.w, blit_rect.h, 8,
             pitch, SDL_PIXELFORMAT_ARGB8888);
     }
 
```
No performance difference, just more clarity.

Reply:
Thanks for the feedback - applied some updates